### PR TITLE
refactor(elements|ino-menu): replace underlying mdc-menu with ino-popover

### LINKED ISSUE #288: ino-menu: Unable to change positioning
### Is your feature request related to a problem? Please describe.

By default, the ino-menu component positions itself. If this functionality does not really work (e.g. outside the viewport), there should be an option to set it manually.

image

### Describe the solution you'd like

A property that takes a corner (e.g. `TOP_LEFT`) and positions the menu surface at that position.

### Describe alternatives you've considered

I tried setting the menu in angular by myself but the mdc-menu seems to apply some inline styles which are hard to overwrite.

### PR DESCRIPTION
**Pull request by janivo**
Closes #288 

## Proposed Changes

- Remove `ino-open` prop
- Use `<ino-popover>` as base
- Use simple trigger mechanism of `<ino-popover>` to open/close the menu
- Use parent as the anchor element
- Add autofocus prop

### CODE REVIEW

**Comment by janivo** [comment]
BREAKING CHANGE:

The `ino-menu` has been updated to use the new `ino-popover` as its underlying component because the `mdc-menu` did not provide the expected customizability.

- Removed property `ino-for`. The parent element will now act as the anchor element
- Removed property `ino-open`. The menu is now a controlled component that handles the open/close state itself. The menu will open/close on anchor click.
- Removed event `menuClosed`. This event is no longer needed because of its controlled behavior.

**Review by pfecht — APPROVED** [review]

File: packages/elements/src/components/ino-menu/ino-menu.tsx [hunk 1]
```diff
@@ -22,85 +17,70 @@ import { MDCCustomMenu } from './MDCCustomMenu';
...
-
-  @Watch('inoFor')
-  inoForChanged() {
-    this.setAnchor(this.inoFor);
-  }
+  @Prop() inoPlacement: Placement = 'auto';
 
   /**
-   * Set this option to show the menu.
+   * If enabled, focuses the first `<ino-list-item>` on menu opening
```
[the remark this answers is not in the document]
Comment (janivo) [reply packages/elements/src/components/ino-menu/ino-menu.tsx:30]: So the consumer has to define it like this:
```html
<ino-button>
  <ino-menu>
    <ino-list-item [data-ino-focus]></ino-list-item>
    <ino-list-item ...></ino-list-item>
  </ino-menu>
</ino-button>
```
And you would search for this attribute on mount?

[the remark this answers is not in the document]
Comment (pfecht) [reply packages/elements/src/components/ino-menu/ino-menu.tsx:30]: Yes. the `[]` are only for the querySelector. The proper attribute would be `<ino-list-item data-ino-focus>`

[the remark this answers is not in the document]
Comment (janivo) [reply packages/elements/src/components/ino-menu/ino-menu.tsx]: Have a look at it again. Is this how you implement it?
If so, we should use this logic on every component which uses a focus prop.

[the remark this answers is not in the document]
Comment (pfecht) [reply packages/elements/src/components/ino-menu/ino-menu.tsx]: Me like. Just keep in mind that the element has to be focussable (buttons, etc.) or it requires a tabindex=-1.

**Review by pfecht — APPROVED** [review]

**Pull request merged**